sdk/servicebus/azure-servicebus/azure/servicebus/aio/management/_management_client_async.py

```diff
@@ -131,6 +132,22 @@ async def _get_rule_element(self, topic_name, subscription_name, rule_name, **kw
             )
         return element
 
+    async def _create_forward_to_header_tokens(self, entity, kwargs):
+        """forward_to requires providing a bearer token in headers for the referenced entity."""
+        kwargs['headers'] = kwargs.get('headers', {})
+
+        async def _populate_header(uri, header):
+            token = (await self._credential.get_token(uri)).token.decode()
+            if not isinstance(self._credential, (ServiceBusSASTokenCredential, ServiceBusSharedKeyCredential)):
+                token = "Bearer {}".format(token)
+            kwargs['headers'][header] = token
+
+        if entity.forward_to:
+            await _populate_header(entity.forward_to, SUPPLEMENTARY_AUTHORIZATION_HEADER)
+        if entity.forward_dead_lettered_messages_to:
+            await _populate_header(entity.forward_dead_lettered_messages_to, \
+                DEAD_LETTER_SUPPLEMENTARY_AUTHORIZATION_HEADER)
```

Comment:
nit picky: if both `forward_to` and `forward_dead_lettered_messages_to` are specified, then 
`_populate_header` would be called twice which means two tokens would be generated and the latter one would overwrite the former one or this is by design?

Reply:
DEAD_LETTER_SUPPLEMENTARY_AUTHORIZATION_HEADER vs SUPPLEMENTARY_AUTHORIZATION_HEADER

They'd both be present, unless I've misunderstood your question.

Reply:
OOB discussion: make it more clear?